query/src/main/java/org/infinispan/query/stats/impl/LocalIndexStatistics.java

```diff
@@ -71,8 +72,15 @@ private CompletionStage<IndexInfo> indexInfos(SearchIndexedEntity indexedEntity)
             () -> session.search(scope).where(SearchPredicateFactory::matchAll).fetchTotalHitCount(), this);
 
       return countStage
-            .thenCompose(count -> indexedEntity.indexManager().unwrap(LuceneIndexManager.class).computeSizeInBytesAsync()
-            .thenApply(size -> new IndexInfo(count, size)));
+            .thenCompose(count -> getLongCompletionStage(indexedEntity, searchMapping)
+                  .thenApply(size -> new IndexInfo(count, size)));
+   }
+
+   private static CompletionStage<Long> getLongCompletionStage(SearchIndexedEntity indexedEntity, SearchMapping searchMapping) {
+      if (searchMapping.isReloading()) {
+         return CompletableFuture.completedFuture(0L);
```

Comment:
Should this be -1 to signal that it is unavailable instead of 0?

Reply:
Good question. I thought to return 0 since we're pretty sure at this stage that either the index does not exist or it does not contain any entity. Thus I think that return 0 is not a trick for the user for this very case. Returning -1 would mean I don't know anything about it, and it is not really true I think.